### DISCUSSION TITLE
TinyIRC improvements

### DISCUSSION THREAD

**Discussion by toncho11** [jbruchon/elks#1571] [General]
This is a list of things I want to improve for tinyIRC:

* set the default server to something that works nowadays
* warn the user if he/she is using a hostname instead of ip address
* warn if the port is different from 6667 that encryption is not supported
* update the usage line in tinyirc with an example line that shows how to connect (with a working IRC server)

**Comment by ghaerr** [comment]
Hello @toncho11,

Thanks for the list of improvements. Are you a C programmer? I'd be happy to help guide you through some of these modifications yourself. If not, I'll try to make them (although I don't actually have tinyirc running yet on QEMU).

> set the default server to something that works nowadays

Please specify what you'd like for both the server and port, very easy to change in tinyirc/Makefile.

> warn the user if he/she is using a hostname instead of ip address

What is the issue here, does tinyirc not convert a hostname to IP address? That should also be easy to fix.

> warn if the port is different from 6667 that encryption is not supported

What is the error message that is given now when connecting to a different port? How exactly should the error message be changed?

> update the usage line in tinyirc with an example line that shows how to connect (with a working IRC server)

Please specify what you'd like to see for the usage line.

In general I think tinyirc is a bit too wordy with its copyright notice and lots of asterisks. I would be for removing most of that. (The copyright would remain within the source code, of course).

Thank you!

**Reply by toncho11** [reply]
Yes, I am a C programmer, but not with a deep knowledge of BIOS, IRQs, etc.  I do Python nowadays :) - high level languages.
I will try to do these modifications when I have the time. 

Does name resolution works in ELKS? That is the question. I thought DNS is not available and so you are obliged to use an IP address? 
There is no error message connecting to another port it is just that I want to warn people not to try to use an encrypted connection and then get disappointed that it does not work.

**Comment by ghaerr** [comment]
> Does name resolution works in ELKS? That is the question. I thought DNS is not available and so you are obliged to use an IP address?

DNS was added over a year ago, and all network utilities (except possibly tinyirc) have been updated to allow names as well as IP addresses. The libc function `in_gethostbyname` uses DNS to translate from name to IP address.

> There is no error message connecting to another port it is just that I want to warn people not to try to use an encrypted connection and then get disappointed that it does not work.

After you play with it more, go ahead and either submit a PR or let me know the precise modifications you'd like to see, and we can get them done.

**Reply by toncho11** [reply]
Thank you!
